src/components/Taxonomy/Taxonomy.tsx

```diff
@@ -112,21 +135,27 @@ function isSubArray(item: string[], parent: string[]) {
   return parent.every((n, i) => item[i] === n);
 }
 
+
 // @todo change `flat` into `alwaysOpen` and move it to context
-const Item = ({ item, isFiltering }: { item: TaxonomyItem, isFiltering?: boolean }) => {
+const Item: React.FC<any> = (props: RowProps) => {
+  const { style, data, index } = props; 
+  const item = data(index);
+  const { row: { id, isOpen, childCount, isFiltering, name, path, padding, isLeaf }, toggle, addInside: addChild } = item;
+  
   const [selected, setSelected] = useContext(TaxonomySelectedContext);
   const { leafsOnly, maxUsages, maxUsagesReached, onAddLabel, onDeleteLabel } = useContext(TaxonomyOptionsContext);
-
-  const checked = selected.some(current => isArraysEqual(current, item.path));
-  const isChildSelected = selected.some(current => isSubArray(current, item.path));
-  const hasChilds = Boolean(item.children?.length);
+  
+  const checked = selected.some(current => isArraysEqual(current, path));
+  const isChildSelected = selected.some(current => isSubArray(current, path));
+  const hasChilds = !isLeaf;
   const onlyLeafsAllowed = leafsOnly && hasChilds;
   const limitReached = maxUsagesReached && !checked;
   const disabled = onlyLeafsAllowed || limitReached;
 
-  const [isOpen, open, , toggle] = useToggle(isChildSelected || isFiltering);
-  const onClick = () => leafsOnly && toggle();
-  const arrowStyle = item.children?.length
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_, open] = useToggle(isChildSelected || isFiltering);
```

Comment:
it was used to open branches with selected items or when the first child is about to add, so now it useless :(
(also in search mode, but this can be fixed by `defaultExpanded`)

Reply:
sorry :/